src/mopidy/core/playback.py

```diff
@@ -280,48 +280,36 @@ def pause(self) -> None:
             self.set_state(PlaybackState.PAUSED)
             self._trigger_track_playback_paused()
 
-    def play(  # noqa: C901, PLR0912
+    def play(
         self,
-        tl_track: TlTrack | None = None,
         tlid: int | None = None,
     ) -> None:
-        """Play the given track, or if the given tl_track and tlid is
-        :class:`None`, play the currently active track.
+        """Play a track from the tracklist, specified by the tracklist ID.
 
-        Note that the track **must** already be in the tracklist.
+        Note that the track must already be in the tracklist.
 
-        .. deprecated:: 3.0
+        If no tracklist ID is provided, resume playback of the the currently
+        active track.
+
+        .. versionremoved:: 4.0
             The ``tl_track`` argument. Use ``tlid`` instead.
 
-        :param tl_track: track to play
-        :param tlid: TLID of the track to play
+        :param tlid: Tracklist ID of the track to play
         """
-        if sum(o is not None for o in [tl_track, tlid]) > 1:
-            raise ValueError('At most one of "tl_track" and "tlid" may be set')
+        if tlid is None and self.get_state() == PlaybackState.PAUSED:
+            self.resume()
+            return
 
-        if tl_track is not None:
-            validation.check_instance(tl_track, models.TlTrack)
+        tl_track: TlTrack | None = None
```

Comment:
This makes the for loop else clause redundant. I think that else clause should log that the track couldn't be found. And maybe even return?

Reply:
Ah, yeah, the else clause is a remnant from when the for loop used the variable name `tl_track`, so one had to reset it if the for loop ended without breaking. That can be removed now.

With regards to returning early, there shouldn't be any intentional logic changes here, and as far as I understand from a brief skim of the code, it will skip to the next track if the one you try to play fails...

Reply:
I added logging now.